Remotion/Web/Development.WebTesting.TestSite.Shared/ScreenshotTest.aspx.cs

```diff
@@ -18,15 +18,15 @@
 using Remotion.Web.ExecutionEngine;
 using Remotion.Web.UI.Controls;
 
-namespace Remotion.Web.Development.WebTesting.TestSite
+namespace Remotion.Web.Development.WebTesting.TestSite.Shared
 {
   public partial class ScreenshotTest : WxePage
   {
     protected override void OnInit (EventArgs e)
     {
       base.OnInit(e);
 
-      var node = new WebTreeNode("ItemA", WebString.CreateFromText("A"), "~/Images/SampleIcon.gif") { IsEvaluated = true, IsExpanded = true };
+      var node = new WebTreeNode("ItemA", WebString.CreateFromText("A"), "~/Image/SampleIcon.gif") { IsEvaluated = true, IsExpanded = true };
```

Comment:
The problem with moving those resources is that its actually more complex than that. There are two main problems:

1. Most of the time it is not allowed to set control properties using `<%= %>`. Thus it is necessary to add code to the c# part, which makes splits up the property values and makes the ASPX harder to reader
2. I don't know how to retrieve the physical path from a mapped `res/.../...` path. `Context.Server.MapPath` does not work

I can move all the resources to the shared folder but it probably takes a few hours to move everything and check that stuff is working correctly. So my question is how important it is to us that we don't duplicate the resources. There are already examples where we have duplications, like the `objects` folder in ObjectBinding.WebTesting.

Reply:
Sorry I mixed two different problems. Both are causes by the moving of the resources into the shared project:

1. `<%= %>` can't be used to set properties, which makes it necessary to use the codebehind to add the `CreateResourceUrl` code. 
2. Downloadables (zip file, text file, etc.) for the download tests require the full path of the file, which is now hidden behind a `res/...` virtual path

I question if it is worth moving the five files to the shared folder compared to the required work needed and benefit gained. Yes it is bit cleaner, but the readability takes a hit and if I am not mistaken we won't need the shared setup in the soon future? If we still want to do the clean separation I need a solution for the physical path as I have no clue how to get it.

Reply:
ad 1) / question: readability: I'm not sure its actually less readable than now if we move all the construction of the WebTab/WebMenuItem to code behind. Markup is just more convenient, but we don't have typesafety, etc. It just wasn't necesarry to do it another way. Looking back, code behind would be my preferred way to do it from the start.

ad 1) / worth it: it feels like it's about 2h of work top. And and given all the retrofitting, am not sure we'll actually re-merge it because that would be extra effort. And it will exercise the primary usage scenario, which is code behind. If we do it cleanly, yes, it's worth it.

ad 2) okay, ok. Hmm..how about we just turn those into embedded resources and write the binary to the output via response.OutputStream or response.BinaryWrite(...)? This leaves just the image file.

ad 1a) oh, and I got another idea, if you really want to avoid the rewrite: try a dataurl. If you switch the image to the NullIcon.gif (and run it through an optimizer first), you get a nice, relatively short string: `data:image/gif;base64,R0lGODlhEAAQAPABAAAA/wAAACH5BAUAAAEALAAAAAAQABAAAAIdRIynGe0PgYG0Wiev3nl72n3iMY1faHLlubQJUAAAOw==` and the screenshots only care about it being 16x16 if I'm not mistaken. 

=> I think with 1a) + 2) we get the best of both worlds and a nice cheap solution?